### ISSUE TITLE
Support MSA Auth

### ISSUE THREAD

**Issue by dtzar** [Microsoft/AzureBot#108]
Currently AzureBoth only supports authentication using Azure Active Directory accounts, not MIcrosoft accounts (i.e. Outlook, Hotmail, etc).  We depend on the AuthBot https://github.com/matvelloso/AuthBot, so this functionality enhancement should happen there.  Enhancement would be making auth work with MSA.  

Bugs would be not graceful handling of when you authenticate with MSA account to let people know we don't support MSA. 
1)	MSA is global admin of tenant - you get: "message": "An error has occurred.", "exceptionMessage": "AADSTS65001: The user or administrator has not consented to use the application with ID '88443132-9b52-485d-90ed-49f37efd1c2b'. Send an interactive authorization request for this user and resource.”
2)	MSA is just a user in the directory - you get an infinite spinning wheel like this:
![image](https://cloud.githubusercontent.com/assets/3578906/18156261/d4ca1020-6fc9-11e6-88df-d0a7504d6932.png)

**Comment by aloneguid** [comment]
I have the same problem, and it doesn't work on some AAD accounts also, specifically ones replicated from on premises.

**Comment by dtzar** [comment]
@aloneguid are the AAD accounts replicated from on-premises which don't work global admins by chance?

**Comment by aloneguid** [comment]
@dtzar no, they aren't definitely. It works randomly on some subscriptions

**Comment by nzthiago** [comment]
At the moment with the current code there's no workaround, you'd have to create a user in your AAD domain and use that to sign in.
 
One flow you can implement is to ask the user for their subscription ID, then use the subscription ID to determine what tenant the user is in, then finally, sign the user in to that tenant using the /<tenant> endpoint instead of /common.
See:  https://azure.microsoft.com/en-us/documentation/articles/resource-manager-api-authentication/
This would work for @hotmail.com and other consumer Microsoft accounts, but would have to be a version of the bot that's not hosted by Microsoft.
